src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -553,7 +560,7 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
                 UpdateCacheExpirationTime(changeWatcher);
             }
 
-            if (data != null)
+            if (data != null && watchedSettings != null)

```

Comment:
Watched settings can be null when there are no refresh registered settings. We dont need to check for null here.